src/gui/transferlistwidget.cpp

```diff
@@ -757,6 +760,48 @@ void TransferListWidget::editTorrentTrackers()
     trackerDialog->open();
 }
 
+void TransferListWidget::exportTorrent()
+{
+    if (getSelectedTorrents().isEmpty())
+        return;
+
+    auto fileDialog = new QFileDialog(this, tr("Choose folder to save exported .torrents"));
+    fileDialog->setAttribute(Qt::WA_DeleteOnClose);
+    fileDialog->setFileMode(QFileDialog::Directory);
+    fileDialog->setOptions(QFileDialog::ShowDirsOnly);
+    connect(fileDialog, &QFileDialog::fileSelected, this, [this](const QString &dir)
+    {
+        const QVector<BitTorrent::Torrent *> torrents = getSelectedTorrents();
+        if (torrents.isEmpty())
+            return;
+
+        const Path savePath {dir};
+        if (!savePath.exists())
+            return;
+
+        const QString errorMsg = tr("Export .torrent failed. Torrent: \"%1\". Save path: \"%2\". Reason: \"%3\"");
+
+        for (const BitTorrent::Torrent *torrent : torrents)
+        {
+            const Path filePath = savePath / Path(torrent->name() + u".torrent");
+            if (filePath.exists())
+            {
+                LogMsg(errorMsg.arg(torrent->name(), filePath.toString(), tr("A file with the same name already exists")) , Log::WARNING);
+                continue;
+            }
+
+            const nonstd::expected<void, QString> result = torrent->info().saveToFile(filePath);
+            if (!result)
+            {
+                LogMsg(errorMsg.arg(torrent->name(), filePath.toString(), result.error()) , Log::WARNING);
+                continue;
+            }
```

Comment:
Shouldn't you show message box after exporting is done? E.g. "Exporting is done. See log for more info."

Reply:
I added a message box which will inform the user when there are errors.